prebackup/prebackup.go

```diff
@@ -175,39 +205,42 @@ func (p *PreBackup) waitForReady(watcher watch.Interface) error {
 
 			last := p.getLastDeploymentCondition(deployment)
 
-			if last != nil {
-				// if the deadline can't be respected https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#progress-deadline-seconds
-				if last.Type == "Progressing" && last.Status == "False" && last.Reason == "ProgressDeadlineExceeded" {
-					watcher.Stop()
-					return fmt.Errorf("error starting pre backup pod %v: %v", deployment.GetName(), last.Message)
-				}
+			if last != nil && deadlineExceededCondition(last) {
+				return fmt.Errorf("error starting pre backup pod %v: %v", deployment.GetName(), last.Message)
 			}
 
-			// Wait until at least one replica is available and continue
-			if deployment.Status.AvailableReplicas > 0 {
+			if hasAvailableReplica(deployment) {
 				return nil
 			}
 
-			p.Log.Info("waiting for command pod to get ready", "name", deployment.GetName(), "namespace", deployment.GetNamespace())
+			p.Log.Info("waiting for command pod to get ready", "name", deployment.Name, "namespace", deployment.Namespace)
 
 		case watch.Error:
 
 			last := p.getLastDeploymentCondition(deployment)
 
 			if last != nil {
-				return fmt.Errorf("there was an error while starting pre backup pod %v: %v", deployment.GetName(), last.Message)
-
+				return fmt.Errorf("there was an error while starting pre backup pod '%v/%v': %v", deployment.Namespace, deployment.Name, last.Message)
 			}
-			return fmt.Errorf("there was an unknown error while starting pre backup pod %v", deployment.GetName())
+			return fmt.Errorf("there was an unknown error while starting pre backup pod '%v/%v'", deployment.Namespace, deployment.Name)
 
 		default:
-			p.Log.Info("unexpected event during deployment wathc ", "name", deployment.GetName(), "event type", event.Type, "namespace", deployment.GetNamespace())
+			p.Log.Info("unexpected event during deployment watch ", "name", deployment.Name, "event type", event.Type, "namespace", deployment.Namespace)
 		}
 	}
 
 	return nil
 }
 
+func deadlineExceededCondition(last *appsv1.DeploymentCondition) bool {
```

Comment:
to not confuse booleans with conditions
```suggestion
func isDeadlineExceeded(last *appsv1.DeploymentCondition) bool {
```